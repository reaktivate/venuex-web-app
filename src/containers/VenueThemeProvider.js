import React from 'react';
import { ThemeProvider } from 'styled-components';
import merge from 'lodash/merge';

import defaultTheme from '../config/defaultTheme.json';

import { withVenueConfig } from './VenueConfigProvider';


const venueConfigToTheme = ({ theme }) => merge(
    defaultTheme,
    theme
);


const VenueThemeProvider = ({ venueConfig, children }) => (
    <ThemeProvider theme={venueConfigToTheme(venueConfig)}>
        {children}
    </ThemeProvider>
);


export default withVenueConfig(VenueThemeProvider);

export const __forTesting = { VenueThemeProvider };
