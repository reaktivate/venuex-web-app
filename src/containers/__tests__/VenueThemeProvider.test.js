import React from 'react';
import styled from 'styled-components';
import { mount } from 'enzyme';

import { __forTesting } from '../VenueThemeProvider';

const { VenueThemeProvider } = __forTesting;


const Child = styled.div`
    color: ${({ theme }) => theme.colors.primary};
    background-color: ${({ theme }) => theme.colors.secondary};
`;

const venueConfig = { theme: { colors: { primary: '#f0f0f0' } } };


it('VenueThemeProvider', () => {
    expect(mount((
        <VenueThemeProvider venueConfig={venueConfig}>
            <Child />
        </VenueThemeProvider>
    )).find('div')).toMatchSnapshot();
});
