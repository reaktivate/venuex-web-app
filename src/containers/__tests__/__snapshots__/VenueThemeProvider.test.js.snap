// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VenueThemeProvider 1`] = `
.c0 {
  color: #f0f0f0;
  background-color: #555;
}

<div
  className="c0"
/>
`;
